dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/PythonGatewayServer.java

```diff
@@ -391,6 +391,42 @@ public void createUser(String userName,
         return result;
     }
 
+    /**
+     * Get processDefinition by given processDefinitionName name. It return map contain processDefinition id, name, code.
+     * Useful in Python API create subProcess task which need processDefinition information.
+     *
+     * @param userName              user who create or update schedule
+     * @param projectName           project name which process definition belongs to
+     * @param processDefinitionName process definition name
+     */
+    public Map<String, Object> getProcessDefinitionInfo(String userName, String projectName, String processDefinitionName) {
```

Comment:
I found this function is most likely to function `createOrUpdateProcessDefinition` in PythonGatewayServer. Could you add a private function combine your function and function `createOrUpdateProcessDefinition` define?